Remove two Exception-swallowing `scope (failure)`

Geod24 opened the pull request:
We're seeing the assert in `UTXODB.find` triggered in the live system.
However, due to the usage of `scope (failure)`, we don't actually know what the error is.

Geod24 committed:
Remove a superfluous scope failure assert

Geod24 committed:
UTXODB.find: Do not swallow Exception, print it and abort

Ideally we should propagate it, but for the time being let
the user debug the issue.

hewison-chris commented on source/agora/consensus/state/UTXODB.d:
```diff
@@ -83,22 +83,31 @@ package class UTXODB
+            auto results = this.db.execute(
+                "SELECT unlock_height, type, amount, locktype, lock FROM utxo WHERE hash = ?", key);
+
+            foreach (row; results)
```
Is there any reason why we use foreach here and not the following:-
`if (!results.empty && results.oneValue!(byte[]).length != 0)`

Geod24 replied:
Nothing I can think of ATM.

hewison-chris reviewed (approved):
LGTM

kchulj commented on source/agora/consensus/state/UTXODB.d:
```diff
@@ -83,22 +83,31 @@ package class UTXODB
+        {
+            import std.stdio;
+            try writeln("Exception thrown in UTXODB.find: ", exc);
+            catch (Exception exc) { assert(0, "Last chance writeln thrown: Giving up"); }
```
Is it common practice to have a `writeln` `try catch` block inside of `catch`?

Geod24 replied:
Well if `writeln` was `nothrow` we wouldn't need the `try` / `catch`

kchulj reviewed (approved):
LGTM

Geod24 committed:
Set assert handler in main.d

This should have been done as part of PR2019 but was forgotten,
rendering the code essentially dead.

Geod24 commented:
Finally. This was very confusing because the `OnInvalidMemoryOperation` was actually triggered in Faucet (most likely because shutdown is broken), but that wasn't clear to me at the time. Should be G2G.
